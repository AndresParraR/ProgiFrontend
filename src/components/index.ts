export * from './vehicleTotal';
export * from './general';
