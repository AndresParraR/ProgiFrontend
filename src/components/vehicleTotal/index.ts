export { default as TableVehicle } from './TableVehicle.vue';
export { default as FilterVehicle } from './FilterVehicle.vue';
export { default as DialogVehicle } from './DialogVehicle.vue';
