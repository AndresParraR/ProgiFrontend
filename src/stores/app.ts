import { NotifyPosition } from "@/enums";
import type { AppState } from "@/types";
import { defineStore } from "pinia";
import type { QNotifyCreateOptions } from "quasar";

export const useAppStore = defineStore("app", {
  state: (): AppState => ({
    _notify: {
      message: "",
      type: "",
      timeout: 0,
      position: NotifyPosition.TOP_RIGHT,
    },
    _isLoading: false,
  }),
  getters: {
    notify(state) {
      return state._notify;
    },
    isLoading(state) {
      return state._isLoading;
    },
  },
  actions: {
    showNotification(
      message: string,
      type: string,
      timeout: number = 4000,
      position: QNotifyCreateOptions["position"] = NotifyPosition.TOP_RIGHT
    ) {
      this._notify = {
        message,
        type,
        timeout,
        position,
      };
    },
    setLoading(val: boolean) {
      this._isLoading = val;
    },
  },
});
