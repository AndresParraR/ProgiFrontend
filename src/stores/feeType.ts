import { FeeTypeService } from '@/services';
import type { FeeTypeState } from '@/types';
import { defineStore } from 'pinia';

export const useFeeTypeStore = defineStore('FeeType', {
  state: (): FeeTypeState => ({
    _feesType: [],
  }),
  getters: {
    feeType(state) {
      return state._feesType;
    },
  },
  actions: {
    async fetchVehiclesType() {
      const res = await FeeTypeService.getAll();
      this._feesType = res.data.response;
    },
  },
});
