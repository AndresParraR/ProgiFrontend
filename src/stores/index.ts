export * from "./app";
export * from "./vehicleTotal";
export * from "./vehicleType";
export * from "./feeType";
