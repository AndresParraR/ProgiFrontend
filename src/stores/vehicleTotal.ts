import { VehicleTotalService } from "@/services";
import type { CreateVehicleTotalParams, VehicleTotalState } from "@/types";
import { defineStore } from "pinia";

export const useVehiclesTotalStore = defineStore("vehiclesTotal", {
  state: (): VehicleTotalState => ({
    _vehiclesTotals: [],
  }),
  getters: {
    vehiclesTotal(state) {
      return state._vehiclesTotals;
    },
  },
  actions: {
    async fetchVehiclesTotal() {
      const res = await VehicleTotalService.getAll();
      this._vehiclesTotals = res.data.response;
    },
    async createVehiclesTotal(params: CreateVehicleTotalParams) {
      const res = await VehicleTotalService.create(params);
      this._vehiclesTotals.push(res.data.response);
    },
    async updateVehicleTotal(id: number, params: CreateVehicleTotalParams) {
      const res = await VehicleTotalService.update(id, params);
      this._vehiclesTotals[
        this._vehiclesTotals.findIndex((el) => el.id == id)
      ] = res.data.response;
    },
    async deleteVehicleTotal(id: number) {
      const res = await VehicleTotalService.delete(id);
      if (res)
        this._vehiclesTotals = this._vehiclesTotals.filter((el) => el.id != id);
    },
  },
});
