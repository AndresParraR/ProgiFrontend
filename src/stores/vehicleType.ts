import { VehicleTypeService } from '@/services';
import type { VehicleTypeState } from '@/types';
import { defineStore } from 'pinia';

export const useVehicleTypeStore = defineStore('vehicleType', {
  state: (): VehicleTypeState => ({
    _vehiclesType: [],
  }),
  getters: {
    vehicleType(state) {
      return state._vehiclesType;
    },
  },
  actions: {
    async fetchVehiclesType() {
      const res = await VehicleTypeService.getAll();
      this._vehiclesType = res.data.response;
    },
  },
});
