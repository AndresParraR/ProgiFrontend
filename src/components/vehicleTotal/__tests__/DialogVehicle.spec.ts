import { mount } from "@vue/test-utils";
import { DialogVehicle } from "@/components";
import { beforeEach, describe, expect, test, vitest } from "vitest";
import { createTestingPinia } from "@pinia/testing";
import { useVehicleTypeStore } from "@/stores";
import { Quasar } from "quasar";

describe("DialogVehicle Component", () => {
  let wrapper: any = null;

  beforeEach(() => {
    const props = {
      showDialog: true,
      vehicle: {
        vehiclePrice: 200,
        vehicleTypeId: 1,
      },
    };

    const global = {
      plugins: [
        createTestingPinia({
          createSpy: vitest.fn,
        }),
        Quasar,
      ],
    };

    wrapper = mount(DialogVehicle, {
      props,
      global,
    });
  });

  test("Should render correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("Should computed vehicleTypes", () => {
    const vehicleTypeStore = useVehicleTypeStore();

    vehicleTypeStore._vehiclesType = [
      {
        id: 1,
        name: "Common",
      },
    ];

    expect(wrapper.vm.vehicleTypes).toEqual([
      {
        label: "Common",
        value: 1,
      },
    ]);
  });

  test("Should computed feesValues", () => {
    wrapper.vm.vehicleTypeId = 1;
    wrapper.vm.vehiclePrice = 398;

    expect(wrapper.vm.feesValues).toEqual({
      baseFee: 39.8,
      specialFee: 7.96,
      associationFee: 5,
      fixedStorageFee: 100,
    });
  });

  test("Should watch for vehicle prop", async () => {
    await wrapper.setProps({
      vehicle: { vehiclePrice: 501, vehicleTypeId: 1 },
    });

    expect(wrapper.vm.vehiclePrice).toEqual(501);
    expect(wrapper.vm.vehicleTypeId).toEqual(1);
  });

  test("Should watch for vehicle prop null", async () => {
    wrapper.vm.vehiclePrice = 57;
    wrapper.vm.vehicleTypeId = 1;

    await wrapper.setProps({
      vehicle: undefined,
    });

    expect(wrapper.vm.cleanForm).toBeTruthy();
    expect(wrapper.vm.vehiclePrice).toEqual(null);
    expect(wrapper.vm.vehicleTypeId).toEqual(null);
  });

  test("Should onSubmit emit vehicleEdited", async () => {
    await wrapper.setProps({
      vehicle: { vehiclePrice: 501, vehicleTypeId: 1 },
    });

    wrapper.vm.onSubmit();

    expect(wrapper.emitted().vehicleEdited).toBeTruthy();
    expect(wrapper.emitted().close).toBeTruthy();
  });

  test("Should onSubmit emit vehicleAdded", async () => {
    await wrapper.setProps({
      vehicle: undefined,
    });

    wrapper.vm.onSubmit();

    expect(wrapper.emitted().vehicleAdded).toBeTruthy();
    expect(wrapper.emitted().close).toBeTruthy();
  });
});
