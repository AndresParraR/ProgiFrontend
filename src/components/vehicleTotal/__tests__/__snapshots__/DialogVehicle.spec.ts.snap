// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogVehicle Component > Should render correctly 1`] = `
<div
  data-v-app=""
>
  
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;
